yt_dlp/extractor/youtube.py

```diff
@@ -1738,9 +1737,8 @@ def mpd_feed(format_id):
             """
             @returns (manifest_url, manifest_stream_number, is_live) or None
             """
-            if time.time() > expiration_time:
-                with lock:
-                    refetch_manifest(format_id)
+            with lock:
```

Comment:
wouldn't this lock unnecessarily many times? `mpd_feed` is called every time a fragment yielded.

Reply:
it will lock, check time and unlock. Is that an issue? If so, this can be reverted

Reply:
I intentionally duplicated code to reduce number of entering locks, as it may wait for other threads

Reply:
decided not to revert this as the effect caused by unnecessary locks seems relatively small